uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget-absence-types-amount-taken.component.js

```diff
@@ -0,0 +1,122 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components',
+  'leave-absences/shared/components/leave-requests-heatmap.component',
+  'leave-absences/shared/models/leave-request.model'
```

Comment:
what's included in the main file (leave-widget.component.js) are the modules for components and models. If I want to include a specific model/component I must specify it (can't just use `<leave-requests-heatmap></leave-request-heatmap>` for example).

This is good because it doesn't include ALL components or models, just the ones needed by the leave widget.

Also, I'd avoid importing the components or models (dependency) in the main file because if we remove the component that needs the dependency we risk including it even if it's not needed.